packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
     }
   }
 
+  private findWeekCorners(weeks: any) {
+    let weekCorners: any = {};
+    let numberOfWeeks = weeks.length;
+    let { firstDayOfWeek, navigatedDate, dateRangeType } = this.props;
+
+    // Get all days in current month
+    let dateRange = getDateRangeArray(navigatedDate, dateRangeType, firstDayOfWeek);
+
+    //Check to see if second to the last day (of the first week) is within the current month
+    if (weeks[0][DAYS_IN_WEEK - 2].originalDate.getMonth() !== navigatedDate.getMonth()) {
```

Comment:
The weeks already have days and for each day, you already know if the day is in the month, check out the isInMonth property